addons/stock/models/stock_move.py

```diff
@@ -518,6 +520,7 @@ def action_show_details(self):
         if self.picking_id.picking_type_id.show_reserved:
             view = self.env.ref('stock.view_stock_move_operations')
         else:
+            self._update_next_serial_count()
```

Comment:
doesn't seem the right place to do that

Reply:
I moved it at the end of `stock.move` `_action_assign` method:

``` python3
if move.product_id.tracking == 'serial':
    move._update_next_serial_count()
```